### PR TITLE
[IMP] images: clean deteled images from file store server

### PR DESCRIPTION
**Pull request by LucasLefevre**
## Description:

description of this task, what is implemented and why it is implemented that way.

Odoo task ID : [3093071](https://www.odoo.com/web#id=3093071&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -76,20 +78,22 @@ import { getXLSX } from "./xlsx/xlsx_writer";
...
+  readonly mode: Mode;
   /**
    * Any external dependencies your custom plugins or functions might need.
    * They are available in plugins config and functions
    * evaluation context.
    */
-  external: {
+  readonly external: Readonly<{
+    fileStore?: FileStore;
+    loadCurrencies?: () => Promise<Currency[]>;
```
Comment (LucasLefevre) [inline src/model.ts:89]: I'm still not 100% sure about this commit

File: src/types/files.ts [hunk 2]
```diff
@@ -1,13 +1,20 @@
 import { Image } from "./image";
 
+type FilePath = string;
```
Comment (rrahir) [inline src/types/files.ts:3]:
```suggestion
type FilePath = string & {}
```
to avoid the resoliton of the alias ?

Comment (LucasLefevre) [reply src/types/files.ts:3]: VS code displays it un-resolved for me :man_shrugging: 
`(method) FileStore.upload(file: File): Promise<FilePath>`

Comment (rrahir) [reply src/types/files.ts:3]: indeed does not work

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**